File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
private: 
    static bool cmp(vector<int> &a, vector<int> &b) {
            return a[0] < b[0];
    }
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ret;
        sort(intervals.begin(), intervals.end(), cmp);
        int first = intervals[0][0];
        int last = intervals[0][1];

        for (int i = 1; i < intervals.size() ; i++) {

            if (intervals[i][0] <= last && intervals[i][1] >= first) {
                first = min(first, intervals[i][0]);
                last = max(last, intervals[i][1]);
            } else {
                ret.push_back({first, last});
                first = intervals[i][0];
                last = intervals[i][1];
            }
        }

        ret.push_back({first, last});
        return ret;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        lo: int = intervals[0][0]
        hi: int = intervals[0][1]
        ret: List[List[int]] = []
        for i in range(1, len(intervals)):
            if intervals[i][0] <= hi:
                hi = max(hi, intervals[i][1])
            else: 
                ret.append([lo, hi])
                lo = intervals[i][0]
                hi = intervals[i][1]

        ret.append([lo, hi])
        return ret

